expan/core/early_stopping.py

```diff
@@ -106,100 +109,75 @@ def group_sequential(x,
+    http://stackoverflow.com/questions/22284502/highest-posterior-density-region-and-central-credible-region
+    
+    :param posterior_samples: sample of data points from posterior distribution of some parameter
+    :type  posterior_samples: array-like
+    :param credible_mass: the range of credible interval. 0.95 means 95% represents credible interval.
+    :type  credible_mass: float
+
+    :return: corresponding lower and upper bound for the credible interval
+    :rtype:  list[float]
+    """
+
     sorted_points = sorted(posterior_samples)
     ciIdxInc = np.ceil(credible_mass * len(sorted_points)).astype('int')
     nCIs = len(sorted_points) - ciIdxInc
     ciWidth = [0] * nCIs
     for i in range(0, nCIs):
         ciWidth[i] = sorted_points[i + ciIdxInc] - sorted_points[i]
 
-    # ciWidth = np.zaeros(nCIs)
-    # ciWidth[:] = sorted_points[ciIdxInc:] - sorted_points
-
     HDImin = sorted_points[ciWidth.index(min(ciWidth))]
     HDImax = sorted_points[ciWidth.index(min(ciWidth)) + ciIdxInc]
-    return (HDImin, HDImax)
-
-
-def get_or_compile_stan_model(model_file, distribution):
-    """
-    Creates Stan model. Compiles a Stan model and saves it to .pkl file to the folder selected by tempfile module if
-        file doesn't exist yet and load precompiled model if there is a model file in temporary dir.
-    Args:
-        model_file: model file location
-        distribution: name of the KPI distribution model, which assumes a 
-            Stan model file with the same name exists
-    Returns:
-        returns compiled Stan model for the selected distribution or normal distribution
-            as a default option
-    Note: compiled_model_file is the hardcoded file path which may cause some issues in future.
-    There are 2 alternative implementations for Stan models handling:
-        1. Using global variables
-        2. Pre-compiling stan models and adding them as a part of expan project
-        (3). Using temporary files with tempfile module is not currently possible, since it 
-            generates a unique file name which is difficult to track.
-        However, compiled modules are saved in temporary directory using tempfile module 
-        which vary based on the current platform and settings. Cleaning up a temp dir is done on boot.
-    """
-    python_version = '{0[0]}.{0[1]}'.format(sys.version_info)
-    compiled_model_file = tempfile.gettempdir() + '/expan_early_stop_compiled_stan_model_' \
-                          + distribution + '_' + python_version + '.pkl'
-
-    if os.path.isfile(compiled_model_file):
-        sm = pickle.load(open(compiled_model_file, 'rb'))
-    else:
-        sm = StanModel(file=model_file)
-        with open(compiled_model_file, 'wb') as f:
-            pickle.dump(sm, f)
-    return sm
-
-cache_sampling_results = False
-sampling_results = {} # memoized sampling results
+    return [HDImin, HDImax]
```

Comment:
Why a list instead of a tuple?

Reply:
good question! I think we don't need to change to a list here, since we can do zip on list and tuple and it will return the same result.